network/codec/codes.go

```diff
@@ -192,7 +194,23 @@ func InterfaceFromMessageCode(code MessageCode) (interface{}, string, error) {
 	case CodeExecutionReceipt:
 		return &flow.ExecutionReceipt{}, what(&flow.ExecutionReceipt{}), nil
 	case CodeResultApproval:
-		return &flow.ResultApproval{}, what(&flow.ResultApproval{}), nil
+		attestation := flow.NewAttestation(
+			flow.Identifier{},
+			flow.Identifier{},
+			0,
+		)
+		body := flow.NewResultApprovalBody(
+			attestation,
+			flow.Identifier{},
+			crypto.Signature{},
+			crypto.Signature{},
+		)
+		approval := flow.NewResultApproval(
+			body,
+			crypto.Signature{},
+		)
+
+		return &approval, what(&approval), nil
```

Comment:
In this case we want to provide an empty value, which then will be deserialized into from a network message. Effectively, we know that construction of this value is not yet completed. Any additional logic from the constructor functions (e.g. relating to ID caching) only needs to happen when we actually have the full value (i.e. immediately after deserialization). So I think the better approach is to just create a variable that is empty-initialized by default, making it explicit that this is not a fully-initialized value:
```suggestion
		var approval flow.ResultApproval
		return &approval, what(&approval), nil
```

Reply:
thanks, that's a valid point.